primer/src/Primer/App.hs

```diff
@@ -1494,17 +1494,17 @@ checkProgWellFormed p =
       -- ensure that the 'ID' is unique across all modules.
       pure $ checkedProg{progSmartHoles = progSmartHoles p}
 
--- | Construct a new, empty expression
+-- | Construct a new, empty expression (with typecache of TEmptyHole, both synthed and checked)
 newExpr :: MonadFresh ID m => m Expr
 newExpr = do
   id_ <- fresh
-  pure $ EmptyHole (Meta id_ Nothing Nothing)
+  pure $ EmptyHole (Meta id_ (Just $ TCEmb $ TCBoth (TEmptyHole ()) (TEmptyHole ())) Nothing)
 
--- | Construct a new, empty type
```

Comment:
It's not obvious (to me, anyway) what this `fix` commit fixes. I can see that a change has been made to `newType` and `newExpr`, and a corresponding test change, but since presumably the test was already passing before this change, it's all a bit circular.

Can you add a short comment to the commit about what, precisely, this change fixes?

Reply:
Thanks & done. This got me to notice the same problem exists for a new session, which I have also fixed in the same way -- see ee21d0c9